[Feature #881] Allow 1 Instance of OpenCS

marcrbouvier opened the pull request.

marcrbouvier committed:
[Feature #811] Only 1 Instance of OpenCS is Allowed

If another instance of OpenCS is started, then it will terminate. This
is done by creating a QLocalServer with a unique ID. If another
QLocalServer with the same ID attempts to be opened, then the creation
of the QLocalServer will fail and the application will terminate.

marcrbouvier committed:
[Feature #881] Raise OpenCS New Window After Re-execution

The OpenCS startup window of the existing exist is raised when trying to
start a new instance. This is done by the new instance connection to the
existing instance's QLocalServer. Once the connection is established the
existing instance raises the startup window.

marcrbouvier committed:
[Feature #881] Code Formatting

Code formatted to the OpenMW policy.

zinnschlag commented:
Looks good mostly. But shouldn't the server name be something unique to OpenCS? This is a global name, right? Theoretically another application could use it and with the current choice we might get a name collision.

nikolaykasyanov commented:
Something like org.openmw.OpenCS should be fine

marcrbouvier committed:
[Feature #881] Rename QLocalServer

The server name should be more unique to the project so there isn't a
interprocess clash.

marcrbouvier commented:
Good idea. I didn't think about that problem. The change has been made.
